solutions/CSES Traffic Lights.mdx

```diff
@@ -0,0 +1,106 @@
+---
+id: cses-1163
+source: CSES
+title: Traffic Lights
+author: Danh Ta Chi Thanh
+---
+
+In this problem, we are given an empty interval length $x$ starting from 0 to $x$, and there are $n$ points being added to the interval chronologically. We want to find the maximum lenght of the gaps between the points at each step.
+
+## Solution
+
+First you can read and think more about the solution here: 
+[Unofficial Editorial](https://codeforces.com/blog/entry/83295) (and also here: [Sketch](https://codeforces.com/blog/entry/86096?#comment-739122))
+
+I will try to explain it: Let's create a set and a multiset. The set is used for storing the positions of the traffic lights, while the multiset will keep track of the "gaps" between the lights. After each addition, the multiset keeps expanding because more lights are put and your work is just to print "the length of the longest passage without traffic lights after each addition" or the max elements of that multiset which is the last element by default. Note that when placing a new traffic light on the road, that light will split the gap between two $adjacent$ lights into 2 smallers pieces, and you also have to remove the length of that gap in the multiset and add 2 new lenghts to the multiset.

```

Comment:
```suggestion
I will try to explain it: Let's create a set and a multiset. The set will store the positions of the traffic lights, while the multiset will keep track of the "gaps" between the lights. The multiset keeps expanding because more lights are added, and you just need to print the length of the longest passage without traffic lights after each addition (i.e. the max element of that multiset). This element is the last element by default.

Note that when placing a new traffic light on the road, that light will split the gap between two adjacent lights into two smaller pieces, so you also have to remove the length of that gap in the multiset and add two new lengths to the multiset.
```